command/ca/revoke.go

```diff
@@ -0,0 +1,391 @@
+'''
+
+Revoke a certificate using a transparently generated token and configured reason:
+'''
+$ step ca revoke --reason "KeyCompromise" 308893286343609293989051180431574390766
+'''
+
+Revoke a certificate using that same certificate to setup an mTLS connection
+with the CA:
+'''
+$ step ca revoke --crt mike.crt --key mike.key 308893286343609293989051180431574390766
+'''
+
+Revoke a certificate using a transparently generated token:
+'''
+$ step ca revoke "KeyCompromise"
+'''
+
+Revoke a certificate using a token, generated by a provisioner, to authorize
+the request with the CA:
+'''
+$ step ca revoke --token <token> --reason "KeyCompromise" 308893286343609293989051180431574390766
+'''`,
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  "reasonCode",
+				Value: "",
+				Usage: `The <reasonCode> for which the certificate is being revoked.
+If unset, default is Unspecified.
+
+
+: <reasonCode> can be a number from 0-9 or a case insensitive string matching
+one of the following options:
+
+    **Unspecified**
+
+    **KeyCompromise**
+
+    **CACompromise**
+
+    **AffiliationChanged**
+
+    **Superseded**
+
+    **CessationOfOperation**
+
+    **CertificateHold**
+
+    **RemoveFromCRL**
+
+    **PrivilegeWithdrawn**
+
+    **AACompromise**
+`,
+			},
+			cli.StringFlag{
+				Name:  "reason",
+				Usage: `The <reason> (string) for which the cert is being revoked.`,
+			},
+			cli.StringFlag{
+				Name:  "serial",
+				Usage: `The <serial number> (string) of the cert that should be revoked.`,
+			},
+			cli.StringFlag{
+				Name:  "crt",
+				Usage: `The path to the <crt> that should be revoked.`,
+			},
+			cli.StringFlag{
+				Name:  "key",
+				Usage: `The path to the <key> corresponding to the cert that should be revoked.`,
+			},
+			tokenFlag,
+			notBeforeFlag,
+			caURLFlag,
+			rootFlag,
+			offlineFlag,
+			caConfigFlag,
+		},
+	}
+}
+
+func revokeCertificateAction(ctx *cli.Context) error {
+	args := ctx.Args()
+	serial := args.Get(0)
+	crtFile, keyFile := ctx.String("crt"), ctx.String("key")
+	token := ctx.String("token")
+	offline := ctx.Bool("offline")
+
+	// offline and token are incompatible because the token is generated before
+	// the start of the offline CA.
+	if offline && len(token) != 0 {
+		return errs.IncompatibleFlagWithFlag(ctx, "offline", "token")
+	}
+
+	// revokeFlow unifies online and offline flows on a single api.
+	flow, err := newRevokeFlow(ctx, crtFile, keyFile)
+	if err != nil {
+		return err
+	}
+
+	// If crt and key are passed then infer the serial number and certificate
+	// that should be revoked.
+	if len(crtFile) > 0 || len(keyFile) > 0 {
+		if len(crtFile) == 0 {
+			return errs.RequiredWithFlag(ctx, "key", "crt")
+		}
+		if len(keyFile) == 0 {
+			return errs.RequiredWithFlag(ctx, "crt", "key")
+		}
+		if len(token) > 0 {
+			errs.IncompatibleFlagWithFlag(ctx, "crt", "token")
+		}
+		if len(serial) > 0 {
+			errs.IncompatibleFlagWithFlag(ctx, "crt", "serial")
+		}
```

Comment:
Incompatible with example in help: `step ca revoke --crt mike.crt --key mike.key 308893286343609293989051180431574390766`